openforcefield/topology/topology.py

```diff
@@ -694,6 +849,25 @@ def atom_start_topology_index(self):
         # Return cached value
         return self._atom_start_topology_index
 
+    @property
+    def virtual_particle_start_topology_index(self):
```

Comment:
I think it's a very useful method to have. Since virtual sites have a variable particle offset, if I am given a random molecule in the topology, it is impossible to use any heuristic to know the value this property provides. This is mostly due to how we define the ordering of atoms vs vsites. For example, right now if I have some molecule in a topology with indices 2,3,4, it is possible that the vsite indices start at 10,002. I think we should expose this functionality.

It also follows the naming convention of similar API calls.